forui/lib/src/widgets/radio.dart

```diff
@@ -0,0 +1,386 @@
+      (true, false) => (FLabelState.enabled, style.enabledStyle),
+      (false, false) => (FLabelState.disabled, style.disabledStyle),
+      (_, true) => (FLabelState.error, style.errorStyle),
+    };
+
+    return GestureDetector(
+      onTap: enabled ? () => onChange?.call(!value) : null,
+      child: FocusableActionDetector(
+        enabled: enabled,
+        autofocus: autofocus,
+        focusNode: focusNode,
+        onFocusChange: onFocusChange,
+        mouseCursor: enabled ? SystemMouseCursors.click : MouseCursor.defer,
+        child: Semantics(
+          label: semanticLabel,
+          enabled: enabled,
+          checked: value,
+          child: FLabel(
+            axis: Axis.horizontal,
+            state: labelState,
+            style: style.labelStyle,
+            label: label,
+            description: description,
+            error: error,
+            child: Stack(
+              alignment: Alignment.center,
+              children: [
+                Container(
+                  padding: const EdgeInsets.all(2),
+                  decoration: BoxDecoration(
+                    border: Border.all(color: stateStyle.borderColor),
+                    color: stateStyle.backgroundColor,
+                    shape: BoxShape.circle,
+                  ),
+                  child: const SizedBox.square(dimension: 10),
+                ),
+                DecoratedBox(
+                  decoration: BoxDecoration(
+                    color: stateStyle.selectedColor,
+                    shape: BoxShape.circle,
+                  ),
+                  child: AnimatedSize(
+                    duration: style.animationDuration,
+                    curve: style.curve,
+                    child: value ? const SizedBox.square(dimension:9) : const SizedBox.shrink(),
+                  ),
+                ),
+              ],
+            ),
+          ),
+        ),
+      ),
+    );
+  }
+
+  @override
+  void debugFillProperties(DiagnosticPropertiesBuilder properties) {
+    super.debugFillProperties(properties);
+    properties
+      ..add(DiagnosticsProperty('style', style))
+      ..add(StringProperty('semanticLabel', semanticLabel))
+      ..add(FlagProperty('value', value: value))
+      ..add(ObjectFlagProperty.has('onChange', onChange))
+      ..add(FlagProperty('enabled', value: enabled))
+      ..add(DiagnosticsProperty('autofocus', autofocus))
+      ..add(DiagnosticsProperty('focusNode', focusNode))
+      ..add(ObjectFlagProperty.has('onFocusChange', onFocusChange));
+  }
+}
+
+/// A [FRadio]'s style.
+class FRadioStyle with Diagnosticable {
+  /// The duration of the animation when the radio's switches between selected and unselected.
+  ///
+  /// Defaults to `const Duration(milliseconds: 100)`.
+  final Duration animationDuration;
+
+  /// The curve of the animation when the radio's switches between selected and unselected.
+  ///
+  /// Defaults to [Curves.linear].
```

Comment:
Should the default value be set in the primary constructor? Also, it appears to default to easOutCirc instead of linear.